Feat: Add Active Tracing Tech preview documentation (#8217)
Kong/docs.konghq.com: app/konnect/gateway-manager/active-tracing.md

---

> title: Active Tracing - Debugging issues in Dataplanes

Reviewer: Debug Data plane issues and unlock In-Depth Insights with Active Tracing

content_type: reference
---

> The Active Tracing feature allows Control Plane Administrators to initiate "deep tracing" sessions in selected Dataplanes. During an active tracing session, the selected Dataplane records Open Telemetry compatible Traces with very detailed spans covering the entire request and response lifecycle for all requests that match a supplied sampling criteria. These traces are pushed up to Konnect where they can inspected using the Konnect builtin span viewer. No additional instrumentation or telemetry technology is needed for generating these traces. 

Reviewer: Active tracing enables control plane administrators to initiate targeted "deep tracing" sessions in specific data plane nodes. During an active tracing session, the selected data plane generates detailed, OpenTelemetry-compatible traces for all requests matching the sampling criteria. The detailed spans are captured for the entire request/response lifecycle. These traces can be visualized with Konnect's built-in span viewer with no additional instrumentation or telemetry tools.

Konnect's active tracing offers exclusive, in-depth insights that cannot be replicated by third-party telemetry tools. The detailed traces generated during live active tracing session are unique to Kong and provide unparalled visibility into system performance.

> The Active Tracing feature allows Control Plane Administrators to initiate "deep tracing" sessions in selected Dataplanes. During an active tracing session, the selected Dataplane records Open Telemetry compatible Traces with very detailed spans covering the entire request and response lifecycle for all requests that match a supplied sampling criteria. These traces are pushed up to Konnect where they can inspected using the Konnect builtin span viewer. No additional instrumentation or telemetry technology is needed for generating these traces. 

Reviewer 2: ```suggestion
Active tracing enables control plane administrators to initiate targeted "deep tracing" sessions in specific data plane nodes. During an active tracing session, the selected data plane generates detailed, OpenTelemetry-compatible traces for all requests matching the sampling criteria. The detailed spans are captured for the entire request/response lifecycle. These traces can be visualized with Konnect's built-in span viewer with no additional instrumentation or telemetry tools.
```

> These deep traces are only generated during a live Active-Tracing session. These traces cannot be generated with 3rd party telemetry software. "Active Tracing" can have an impact on throughput under heavy load conditions. Active-Tracing itself adds negligible latency to request-response processing under normal load conditions.

Reviewer: Konnect's active tracing offers exclusive, in-depth insights that cannot be replicated by third-party telemetry tools. The detailed traces generated during live active tracing session are unique to Kong and provide unparalled visibility into system performance.

> These deep traces are only generated during a live Active-Tracing session. These traces cannot be generated with 3rd party telemetry software. "Active Tracing" can have an impact on throughput under heavy load conditions. Active-Tracing itself adds negligible latency to request-response processing under normal load conditions.

Reviewer 2: ```suggestion
These deep traces are only generated during a live Active-Tracing session. These traces cannot be generated with 3rd party telemetry software. "Active Tracing" can have an impact on throughput under heavy load conditions. Active-Tracing itself adds negligible latency to request-response processing under normal load conditions.
```

> Where feasible span and attribute naming conventions follow OpenTelemetry naming standards.

Reviewer: Active Tracing adheres to OpenTelemetry naming conventions for spans and attributes, wherever possible, ensuring consistency and interoperability.

> Where feasible span and attribute naming conventions follow OpenTelemetry naming standards.

Reviewer 2: ```suggestion
Active Tracing adheres to OpenTelemetry naming conventions for spans and attributes, wherever possible, ensuring consistency and interoperability.
```

> You can start a tracing session by navigating to Gateway Manager, choosing a Control Plane and selecting the "Active Tracing" menu option. A sample trace can be seen below. It is clear from inspecting the spans that the bulk of the latency occurs in the pre-function "plugin" running in the access phase. 

Reviewer: ### Get started with tracing

Active Tracing requires the following data plane version and environment variables in kong.conf

1. Version 3.9+ 
2. KONG_CLUSTER_RPC=on 
3. KONG_ACTIVE_TRACING=on.

##### To start a tracing session 

1. Navigate to Gateway Manager
2. Select a Control Plane with the appropriate data plane connected 
3. Click on "Start a tracing session", specify an optional name and click start

You should start to see the traces being captured. The default session time is 5min. For advanced sampling rules, specify the criteria

A sample trace is show below. By inspecting the spans, it is clear that bulk of the latency occcurs in the pre-function plugin in access phase

> You can start a tracing session by navigating to Gateway Manager, choosing a Control Plane and selecting the "Active Tracing" menu option. A sample trace can be seen below. It is clear from inspecting the spans that the bulk of the latency occurs in the pre-function "plugin" running in the access phase. 

Reviewer 2: ```suggestion

## Get Started with tracing

Active Tracing requires the following **data plane version** and **environment variables** in `kong.conf`:

- **Version:** 3.9+
- **Environment Variables:**
  - `KONG_CLUSTER_RPC=on`
  - `KONG_ACTIVE_TRACING=on`

## Start a Tracing Session

1. Navigate to **Gateway Manager**.
2. Select a **Control Plane** with an appropriate data plane connected.
3. Click **"Start a tracing session"**, specify an optional name, and click **Start**.

Once started, traces will begin to be captured. The **default session duration** is **5 minutes**.  
For **advanced sampling rules**, specify the desired criteria.

## Sample Trace

A sample trace is shown below. By inspecting the **spans**, it is clear that the **bulk of the latency** occurs in the **pre-function plugin** during the **access phase**.

```

---

> title: Active Tracing - Debugging issues in Dataplanes

Reviewer 2: ```suggestion
title: Debug Data plane issues and unlock In-Depth Insights with Active Tracing
```
